README.md

```diff
@@ -1,4 +1,414 @@
+    url: "/some/endpoint",
+    locale: "en-US"  
+  }
+}
+```
+
+## Should Request
+
+Prevents unnecessary requests by checking state beforehand. Can be used to fetch some resources only once. 
+
+```js
+{
+  type: "some-type",
+  request: {
+    url: "/some/endpoint",
+    shouldRequest(state) {
+     return !state.isFetched;
+    }
+  }
+}
+```
+
+## Abort Callback
+
+Requests can be cancelled as long as they are still running. You receive the cancel function by providing `getAbortCallback` in request options.
+
+```js
+let cancelRequest;
+
+dispatch({
+  type: "some-type",
+  request: {
+    url: "/some/endpoint",
+    getAbortCallback: (callback) => {
+      cancelRequest = callback;
+    }
+  }
+})
+
+
+// later in app
+cancelRequest();
+
+```
+
+## Request Types
+
+`request.type` attribute determines which http method should be used.
+
+### `query`
+
+Sends out GET request. Collects parameters from `request.query` and applies pagination query options if there are any.
+
+```js
+
+{
+  type: "some-type",
+  request: {
+     url: "/users",
+     type: 'query',
+     query: { q: 'Peter' } 
+  }
+}
+```
+
+
+### `delete` / `head` / `options` / `post` / `put` / `patch`
+
+```js
+
+{
+  type: "some-type",
+  request: {
+    url: "/some/endpoint/123",
+    type: 'put',
+    payload: { user: { email, password }}
+  }  
+}
+```
+
+
+## Pagination
+
+There is some boilerplate needed to fully support paginated lists loaded over the network. These helpers help with managing state and sending pagination params over the network.
+
+### `assignPaginationDefaults` / `paginationGenerator` helpers
+
+```js
+import { paginationGenerator, assignPaginationDefaults } from 'nebenan-redux-tools/lib/network/pagination';
+
+const getDefaultState = () => assignPaginationDefaults({ entities: {} });
+const getPaginationUpdates = paginationGenerator(types.BOOKMARKS_FETCH);
+
+export const reducer = (state = getDefaultState(), action) => {
+  let newState = state;
+  
+  const update = getPaginationUpdates(newState, action);
+  if (update) newState = updeep(update, newState);
+  
+  switch (action.type) {
+    case resolved(types.BOOKMARKS_FETCH): {
+      const { result, entities } = parse(action.payload);
+
+      const collection = newState.collection.concat(result.bookmarks);
+
+      return updeep({ collection, entities }, newState);
+    }
+    
+    default: {
+      return newState;
+    }
+  }
+}
+```
+
+The state now holds the following information
+```js
+{
+  currentPage: 0, 
+  // datetime of last fetch
+  lastFetched: 0,
+  // request in progress  
+  isFetching: false,
+  // request failed  
+  isFailed: false,  
+  // total amount of items over all pages
+  total: null,
+  // item ids for current page  
+  collection: [],
+}
+```
+
+### Actions
+
+Pagination query params will be added for `query` requests with either `request.pagination.first` or `request.pagination.last` set.
+
+*Only works for endpoints supporting `lower`/`higher` params.*
```

Comment:
do we need to document how to increase page parameter in the default pagination?

Reply:
The page parameter is updated by `paginationGenerator`. No need to do anything